### PR TITLE
Many improvements to everything

### PR DESCRIPTION
**Pull request by RealMuffinTime**
Hello there, this PR does some things, mostly to convert it into a usable state:

- Allow the mod to build :)
- Update/add many strings
- Catch invalid URLs
- Improve config usage
- Write inital config file if not exisitent
- Remove old secrets.txt reference

And the most relevant changes, how messages are sent:
- Add an icon of player to their messages
- Make join/leave/death messages look like send from player (with icon and name)
- Add a player `counter current/max` to join/leave messages
- Change startup and shutdown messages 

NVM ~~Altough I did some changes/improvents to config saving, I would recommend to rather use something like [cloth-config-api](https://modrinth.com/mod/cloth-config).~~
This PR took inspiration of the fork [LittleHooksCustom](https://github.com/RealMuffinTime/LittleHooksCustom) of [LittleHooks](https://github.com/Pequla/LittleHooks).

### CODE REVIEW

**Comment by AV306** [comment]
Wow, this is great!
Thanks for the improvements, I'll merge this now :D

**Comment by AV306** [comment]
(I chose not to use Cloth Config because its main feature is config GUIs which are useless on server-side, and I wanted ChatHook to have no dependencies)

File: src/main/java/me/av306/chathook/minecraft/ChatHook.java [hunk 1]
```diff
@@ -38,54 +39,79 @@ public enum ChatHook
...
             (message, source, params) ->
             {
                 if ( this.logCommandMessages && this.enabled )
-                    WebhookSystem.INSTANCE.sendMessage( source.getName(), message.getSignedContent() );
+                    WebhookSystem.INSTANCE.sendMessage( null, message.getSignedContent() );
             }
         );
 
+        // Server startup process
+        ServerLifecycleEvents.SERVER_STARTING.register(
```
Comment (AV306) [inline src/main/java/me/av306/chathook/minecraft/ChatHook.java:103]: I think we only need one message when the server has finished starting, so this is unnecessary.

File: src/main/java/me/av306/chathook/webhook/WebhookSystem.java [hunk 2]
```diff
@@ -13,66 +9,52 @@
 import java.net.http.HttpResponse.BodyHandlers;
 
 import me.av306.chathook.minecraft.ChatHook;
+import net.minecraft.server.network.ServerPlayerEntity;
 
 public enum WebhookSystem
 {
     INSTANCE;
-    
-    private final URI WEBHOOK_URI;
```
Comment (AV306) [inline src/main/java/me/av306/chathook/webhook/WebhookSystem.java:21]: Why do you want to call into the config manager's HashMap and create a new URI object every time a message is sent? I suspect that the compiler may not be able to optimise the call(s) out.

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,3 @@
-# secret
-run/secrets.txt
```
Comment (AV306) [inline .gitignore:2]: Why?

File: src/main/java/me/av306/chathook/minecraft/ChatHook.java [hunk 4]
```diff
@@ -38,54 +39,79 @@ public enum ChatHook
...
-        // Game messages e.g. join/leave game messages, death messages
-        ServerMessageEvents.GAME_MESSAGE.register(
-            (server, message, overlay) ->
-            {
-                if ( this.logGameMessages && this.enabled )
-                    WebhookSystem.INSTANCE.sendMessage( "Server", message.getString() );
-            }
+        // Player joined the server
+        ServerPlayConnectionEvents.JOIN.register(
+                (sender, player, hand) -> {
```
[the remark this answers is not in the document]
Comment (RealMuffinTime) [reply src/main/java/me/av306/chathook/minecraft/ChatHook.java:70]: Yeah, I don't know how that happened, also applies to the other registers, having also nonsensical variable names.

File: src/main/java/me/av306/chathook/minecraft/ChatHook.java [hunk 5]
```diff
@@ -38,54 +39,79 @@ public enum ChatHook
...
         );
 
-        // Command messages, supposedly.
-        // FIXME: doesn't seem to work
+        // Player left the server
+        ServerPlayConnectionEvents.DISCONNECT.register(
+                (sender, player) -> {
+                    if ( this.logGameMessages && this.enabled )
+                        WebhookSystem.INSTANCE.sendMessage( sender.player,
+                                String.format("**%s left the game** %d/%d",
```
[the remark this answers is not in the document]
Comment (RealMuffinTime) [reply src/main/java/me/av306/chathook/minecraft/ChatHook.java:85]: I generally would say that the current format is pretty clear.
I think the ` - %d players online` would dominate to much. Having the max player count is subtle and makes everyone understand (I think.), that this number regarding online players.
grafik
vs
grafik

File: src/main/java/me/av306/chathook/minecraft/ChatHook.java [hunk 6]
```diff
@@ -38,54 +39,79 @@ public enum ChatHook
...
+        );
+
+        // Command messages
+        // It does work, it just posts messages originating from e.g. '/say message'
         ServerMessageEvents.COMMAND_MESSAGE.register(
             (message, source, params) ->
             {
                 if ( this.logCommandMessages && this.enabled )
-                    WebhookSystem.INSTANCE.sendMessage( source.getName(), message.getSignedContent() );
+                    WebhookSystem.INSTANCE.sendMessage( null, message.getSignedContent() );
```
[the remark this answers is not in the document]
Comment (RealMuffinTime) [reply src/main/java/me/av306/chathook/minecraft/ChatHook.java:98]: When leaving at `null` Discord uses the webhook configured settings.
That way you can give the webhook an icon and a name. As example the server name and icon.
And the source in this case is pretty much always the server.

File: src/main/java/me/av306/chathook/minecraft/ChatHook.java (hunk 1 above)
Comment (RealMuffinTime) [reply src/main/java/me/av306/chathook/minecraft/ChatHook.java:103]: Sure. Also an config option would be possible.

File: src/main/java/me/av306/chathook/webhook/WebhookSystem.java (hunk 2 above)
Comment (RealMuffinTime) [reply src/main/java/me/av306/chathook/webhook/WebhookSystem.java:21]: You can change the webhook url on the fly via a command. So maybe thats why. Idk.

**Pull request merged**